src/azure-cli/azure/cli/command_modules/synapse/commands.py

```diff
@@ -137,5 +169,84 @@ def get_custom_sdk(custom_module, client_factory):
+        g.custom_command('set', 'create_or_update_linked_service', supports_no_wait=True)
+        g.custom_command('list', 'list_linked_service')
+        g.custom_show_command('show', 'get_linked_service')
+        g.custom_command('delete', 'delete_linked_service', confirmation=True, supports_no_wait=True)
+
+    # Data Plane Commands --Artifacts dataset operations
+    with self.command_group('synapse dataset', synapse_dataset_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('create', 'create_or_update_dataset', supports_no_wait=True)
+        g.custom_command('set', 'create_or_update_dataset', supports_no_wait=True)
+        g.custom_command('list', 'list_datasets')
+        g.custom_show_command('show', 'get_dataset')
+        g.custom_command('delete', 'delete_dataset', confirmation=True, supports_no_wait=True)
+
+    # Data Plane Commands --Artifacts pipeline operations
+    with self.command_group('synapse pipeline', synapse_pipeline_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('create', 'create_or_update_pipeline', supports_no_wait=True)
+        g.custom_command('set', 'create_or_update_pipeline', supports_no_wait=True)
+        g.custom_command('list', 'list_pipelines')
+        g.custom_show_command('show', 'get_pipeline')
+        g.custom_command('delete', 'delete_pipeline', confirmation=True, supports_no_wait=True)
+        g.custom_command('create-run', 'create_pipeline_run')
+
+    # Data Plane Commands --Artifacts pipeline run operations
+    with self.command_group('synapse pipeline-run', synapse_pipeline_run_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('query-by-workspace', 'query_pipeline_runs_by_workspace')
+        g.custom_show_command('show', 'get_pipeline_run')
+        g.custom_command('cancel', 'cancel_pipeline_run', confirmation=True)
+
+    with self.command_group('synapse activity-run', synapse_pipeline_run_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('query-by-pipeline-run', 'query_activity_runs')
+
+    # Data Plane Commands --Artifacts trigger operations
+    with self.command_group('synapse trigger', synapse_trigger_sdk,
+                            custom_command_type=get_custom_sdk('artifacts', None)) as g:
+        g.custom_command('create', 'create_or_update_trigger', supports_no_wait=True)
+        g.custom_command('set', 'create_or_update_trigger', supports_no_wait=True)
+        g.custom_command('list', 'list_triggers')
+        g.custom_show_command('show', 'get_trigger')
+        g.custom_command('delete', 'delete_trigger', confirmation=True, supports_no_wait=True)
+        g.custom_command('subscribe-to-event', 'subscribe_trigger_to_events', supports_no_wait=True)
```

Comment:
For simplicity, using "subscribe" as command is enough? also for "unsubscribe".

Reply:
It keeps align with ADF too.

Reply:
I don't think ADF uses good command name, we should do right thing, but not keep align with bad thing.

---
In reply to: [506171544](https://github.com/Azure/azure-cli/pull/15296#discussion_r506171544) [](ancestors = 506171544)